src/app/cours/cours.component.ts

```diff
@@ -12,45 +14,147 @@ import { Unit } from '../model/unit';
 })
 
 export class CoursComponent implements OnInit {
-  myClasses: Cours[] = [];
-  classForm: FormGroup;
 
-  constructor(private coursService: CoursService, private formBuilder: FormBuilder) {
-    this.classForm = this.formBuilder.group({
-      label: ['', Validators.required],
-      period: ['', Validators.required],
-      nbHour: ['',Validators.required],
-      teacher: ['', Validators.required],
-      detail: ['', Validators.required]
+  myClasses: Cours[] = []; 
+  classForm: FormGroup;  
+  profs: Professeur [];
+
+  constructor(private coursService: CoursService, 
+              private profService: ProfService) {
+                
+    this.classForm = new FormGroup({
+      label: new FormControl('', Validators.required),
+      period: new FormControl('', Validators.required),
+      hours: new FormControl('',Validators.required),
+      professor: new FormControl('', Validators.required),
+      description: new FormControl('', Validators.required)
     });
   }
 
-  ngOnInit() {
-    this.myClasses = this.coursService.getCours();
+  ngOnInit(): void {
+    // essayer d'utiliser le forkJoin de RXJS
+    //https://www.learnrxjs.io/learn-rxjs/operators/combination/forkjoin
+   this.getCours();
+   this.getProfs();
+  }
+
+  getCours(): void{
+     this.coursService.getCours()
+    .subscribe(
+      (data)=> {
+        this.myClasses = data;
+      },
+      (err)=> {
+        Swal.fire(
+        'Attention',
+        'Une erreur c\'est produit lors du chargement des cours',
+        'warning');
+      }
+    )
+  }
+
+  getProfs(){
+    this.profService.getProfs()
+    .subscribe(
+      (data)=> {
+        this.profs = data;
+      },
+      (err)=> {
+        Swal.fire(
+          'Attention',
+          'Une erreur c\'est produit lors du chargement des professeurs',
+          'warning'  
+        );}
+    ) 
   }
 
   deleteAll(): void {
-    //this.myClasses = this.coursService.deleteAllClasses();
-    this.myClasses = [];
+    this.myClasses =[];
   }
+  //essayer de créer des sous fonctions pour réduire la taille de la fonction
+  deleteClass(id: number, coursName: string): void {
```

Comment:
too long function try to move all the Swal (alerts) functions into another file and import it here.
Like create an AlertManager with some functions like warning and error...